drainer/sync/pb.go

```diff
@@ -24,21 +28,44 @@ import (
 var _ Syncer = &pbSyncer{}
 
 type pbSyncer struct {
-	binlogger binlogfile.Binlogger
-
 	*baseSyncer
+
+	binlogger binlogfile.Binlogger
+	cancel    func()
 }
 
 // NewPBSyncer sync binlog to files
-func NewPBSyncer(dir string, tableInfoGetter translator.TableInfoGetter) (*pbSyncer, error) {
-	binlogger, err := binlogfile.OpenBinlogger(dir, binlogfile.SegmentSizeBytes)
+func NewPBSyncer(cfg *DBConfig, tableInfoGetter translator.TableInfoGetter) (Syncer, error) {
```

Comment:
- return the type instead of the interface
- pass the specified param you need

Reply:
Would you mind explaining why?

Reply:
- return the type instead of the interface
it will be flexible to use pBSyncer directly if need? 
- pass the specified param you need
decouple it from `DBConfig` and the redundant `DBConfig` contains too many fields unrelated to PBSyncer, user will never know how to use it actually unless reading the code of `NewPBSyncer`.

Reply:
> * return the type instead of the interface
>   it will be flexible to use pBSyncer directly if need?
https://github.com/golang/go/wiki/CodeReviewComments#interfaces
Go interfaces generally belong in the package that uses values of the interface type, not the package that implements those values. The implementing package should return concrete (usually pointer or struct) types: that way, new methods can be added to implementations without requiring extensive refactoring.
> * pass the specified param you need
>   decouple it from `DBConfig` and the redundant `DBConfig` contains too many fields unrelated to PBSyncer, user will never know how to use it actually unless reading the code of `NewPBSyncer`.